cloudpathlib/local/localclient.py

```diff
@@ -49,6 +49,10 @@ def get_default_storage_dir(cls) -> Path:
     @classmethod
     def reset_default_storage_dir(cls) -> Path:
         cls._default_storage_temp_dir = None
+
+        # Also reset default client so it gets recreated with new temp dir
+        cls._default_client = None
```

Comment:
I'm not convinced this is the right fix. 

It took me a while to remember what `reset_default_storage_dir` is supposed to do and why, and I had to [look at our docs](https://cloudpathlib.drivendata.org/stable/testing_mocked_cloudpathlib/#2-get-a-pathlibpath-object-that-points-to-the-local-storage-directory).

It feels weird to me that resetting the storage directory has a side effect of changing the default client. If someone had set a default client with non-default values (e.g., say they set a non-default `file_cache_mode`), then resetting the default client here would lead to unexpected changes later.

It kind of seems to me like the more sensible design is that the storage directory shouldn't be a class instance at all, and that it should be a client instance attribute (i.e., setting storage directory to a default value should be a method and not a property or attribute). And then for cleanup, rather than a `reset_default_storage_dir` method, there should be a `reset_default_client` method.